radial_build_tools.py

```diff
@@ -0,0 +1,474 @@
+        regions = {}
+
+        regions["plasma"] = -self.surfaces["plasma_surface"]
+
+        surf_list = list(self.surfaces.keys())
+
+        for inner_surf, outer_surf in zip(surf_list[0:-1], surf_list[1:]):
+            regions[outer_surf] = (
+                -self.surfaces[outer_surf] & +self.surfaces[inner_surf]
+            )
+
+        self.regions = regions
+        self.surf_list = surf_list
+
+    def build_cells(self):
+        """
+        Build OpenMC cells from the regions defined by the build dict
+        """
+        # build cells
+        cell_dict = {}
+        materials = set()
+
+        cell_dict["plasma_cell"] = openmc.Cell(
+            region=self.regions["plasma"], name="plasma_cell"
+        )
+
+        for layer, layer_def in self.build.items():
+            if layer_def["thickness"] != 0:
+                cell_dict[layer] = openmc.Cell(
+                    region=self.regions[layer],
+                    name=layer,
+                    fill=layer_def["material"],
+                )
+                materials.add(layer_def["material"])
+
+        self.cell_list = list(cell_dict.values())
+        self.cell_dict = cell_dict
+        self.materials = materials.discard(None)
+
+    def get_bounded_geometry(self):
+        """
+        Get an OpenMC geometry instances containing all cells, plus a bounding
+        vacuum cell
+        """
+        unbounded_geometry = openmc.Geometry(self.cell_list)
+
+        bounding_box = unbounded_geometry.bounding_box
+
+        vac_surf = openmc.Sphere(
+            r=np.sum(
+                np.multiply(
+                    (bounding_box[1] - bounding_box[0]),
+                    (bounding_box[1] - bounding_box[0]),
+                )
+            )
+            ** 0.5,
+            boundary_type="vacuum",
+        )
+
+        vac_region = -vac_surf & +self.surfaces[self.surf_list[-1]]
+        vac_cell = openmc.Cell(region=vac_region, name="vac_cell")
+
+        self.cell_list.append(vac_cell)
+        self.cell_dict["vac_cell"] = vac_cell
+
+        self.geometry = openmc.Geometry(self.cell_list)
+
+    def build_openmc_model(self):
+        """
+        Builds openmc model using the build definition
+        """
+        self.build_surfaces()
+        self.build_regions()
+        self.build_cells()
+        self.get_bounded_geometry()
+
+    def get_openmc_model(self):
+        """
+        Return toroidal model built using the build definition, contains
+        geometry and material information
+
+        Returns:
+            model (openmc model): Model containing materials and geometry
+                from the build dict.
+            cells (dict): dict mapping layer names to openmc cell instances in
+                the model object returned by this function.
+        """
+        self.build_openmc_model()
+        model = openmc.Model(geometry=self.geometry, materials=self.materials)
+        return model, self.cell_dict
+
+    def get_radial_build_plot(
```

Comment:
fixed in https://github.com/Edgar-21/radial_build_tools/pull/1

Reply:
Sorry if it was more confusing than helpful to make a new branch and PR on top of this one, maybe I should have moved it to this repo first instead of it being on my fork? I could just merge it now as well.

Reply:
I'll just make the changes suggested in this pr on this branch, and then PR the new features once this one gets finished up.

Reply:
You don't need to PR changes to your own branch, you can just push them...